### ISSUE TITLE
Code blocks broken on iOS

### ISSUE THREAD

**Issue by tatupesonen** [fasterthanlime/feedback#59]
**Where's the problem at?**
Please link to the article

https://fasterthanli.me/articles/a-rust-match-made-in-hell

**No, seriously, where is it?**
Certain code blocks are truncated on the right side when using Safari on latest iOS.

**Describe the issue and a fix if you know it**

![image](https://user-images.githubusercontent.com/7150217/167685818-3e87139c-0e0b-4bea-8994-fcaf21e7d895.jpeg)
![image](https://user-images.githubusercontent.com/7150217/167685832-3e2adbcf-e481-4e44-b434-98d7bcedb226.jpeg)
![image](https://user-images.githubusercontent.com/7150217/167685845-54242fb9-24f6-49fc-8739-3d189bd6d817.jpeg)

**Comment by tatupesonen** [comment]
I don't know why that auto assignment happened. I created this issue on mobile, so sorry about that.

**Comment by fasterthanlime** [comment]
> I don't know why that auto assignment happened. I created this issue on mobile, so sorry about that.

No worries, I'm the only maintainer anyway 😅

**Comment by remmycat** [comment]
I looked into this a little bit and it seems that the rendering of a `pre` tag has some weird behaviour in Safari (mobile and desktop).

The "problem" seems to be that the `.code-inner` class (aka the `pre`) has `overflow-x: hidden` set.

For overflow hidden, browsers afaik shrink the content width as much as they can, with a default minimum of `min-content`, and cut off anything that is larger than that.

Consider this example code of your recent article.
```rust
impl Display for Either {
    delegate::delegate! {
        to self.display() {
            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result;
        }
    }
}
```

As it's just a bunch of text and inline elements, without width shenanigans going on, the `min-content` width of this should really just be the width of the longest line of the rendered text.

Now this is where it gets funky in Safari:

Safari's `min-content` width calculation seems to get messed up *a lot* on your site.

When manually setting `width: min-content` on the `pre` this happens:
<img width="363" alt="Screenshot 2022-06-03 at 14 03 42" src="https://user-images.githubusercontent.com/3317423/171849909-0f2fce0b-015d-407e-84ce-38211a0cae3d.png">

I couldn't really pinpoint what causes this - but it somehow seems to be related to whitespace of text nodes next to the `<i>` elements of the syntax highlighting. This is probably a browser bug (still existant in Safari TP), but I don't have the spoons to report it 🙃

My best guess is that internally the layouting thinks that the content can wrap, even though visually it won't.
In the past there were bugs in Safari related to the `pre` content wrapping when it shouldn't.

The only good workaround I found is
- Set `min-width: max-content` on `.code-inner` `pre`s.
